File: ui/react/src/action/ExecuteQuery.ts
import Parameter from "../model/Parameter";
import TableAction from "./TableAction";
import Action from "./Action";

class ExecuteQuery extends TableAction {
    public params: Parameter[];

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const params = [];
        if (typeof input.params !== 'undefined') {
            for (const param of input.params) {
                params.push(Parameter.deserialize(param));
            }
        }
        return new ExecuteQuery(sheetName, tableName, params);
    }

    constructor(sheetName: string, tableName: string, params: Parameter[]) {
        super(sheetName, tableName);
        this.params = params;
    }

    public wrapper(): Action {
        const action = new Action();
        action.executeQuery = this;
        return action;
    }

}

export default ExecuteQuery;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/ExecuteQuery.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.variant.Variant;

import java.util.Map;

@Deprecated
public class ExecuteQuery extends TableAction {
    private Map<String, Variant> params;

    public ExecuteQuery() {
    }

    public ExecuteQuery(String sheetName, String tableName, Map<String, Variant> params) {
        super(sheetName, tableName);
        this.params = params;
    }

    public Map<String, Variant> getParams() {
        return params;
    }

    public void setParams(Map<String, Variant> params) {
        this.params = params;
    }
}
